shared/program_test.go

```diff
@@ -87,3 +89,54 @@ func TestProgramMarshalUnmarshalJSON(t *testing.T) {
 		t.Errorf("Operations length mismatch after roundtrip")
 	}
 }
+
+func TestProgramMarshalWithOpsMask(t *testing.T) {
```

Comment:
Remove this test. Extend the TestNewProgramFromText_XXX tests to check the extracted OpsMasks.